compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp

```diff
@@ -159,6 +159,95 @@ static LogicalResult setRootConfigForPackPipeline(func::FuncOp entryPointFn,
   return success();
 }
 
+template <typename ADDOP, typename MULOP>
+static bool bodyMatcherForMatmulTranspose(Value yieldVal, Block *body) {
+  auto addOp = yieldVal.getDefiningOp<ADDOP>();
+  if (!addOp) return false;
+  auto mulOp = addOp.getOperand(1).template getDefiningOp<MULOP>();
+  if (!mulOp) return false;
+  auto lhsBlockArg = mulOp.getOperand(0).template dyn_cast<BlockArgument>();
+  auto rhsBlockArg = mulOp.getOperand(1).template dyn_cast<BlockArgument>();
+  auto outBlockArg = addOp.getOperand(0).template dyn_cast<BlockArgument>();
+  if (!lhsBlockArg || !rhsBlockArg || !outBlockArg ||
+      lhsBlockArg.getOwner() != body || rhsBlockArg.getOwner() != body ||
+      outBlockArg.getOwner() != body || lhsBlockArg.getArgNumber() != 0 ||
+      rhsBlockArg.getArgNumber() != 1 || outBlockArg.getArgNumber() != 2) {
+    return false;
+  }
+  return true;
+}
+
+/// `isMatmulTranspose` is a utility function that aims to indentify whether a
+/// linalg.generic op is a matmul transpose op.
+static bool isMatmulTranspose(linalg::GenericOp genericOp) {
```

Comment:
Instead of using a `linalg.generic` here, you can use a `LinalgOp` directly. Then whether you have a named op or a `linalg.generic` that implements the same operation, it will use the same matcher.